torch/utils/cpp_extension.py

```diff
@@ -66,6 +66,23 @@ def _find_cuda_home():
 
                               !! WARNING !!
 '''
+WRONG_COMPILER_WARNING = '''
+
+                               !! WARNING !!
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+Your compiler ({0}) is not compatible with the compiler Pytorch was built with
+for this platform, which is {1} on {2}. Please use {1} to to compile your
+extension. Alternatively, you may compile PyTorch from source using {0}, and
+then you can also use {0} to compile your extension.
```

Comment:
Can we use names instead of indices? Otherwise, the mapping should be at least documented.

Reply:
I know, and `format()` supports keyword arguments.

Reply:
I see. I think it's a good idea and I would do it but it's just about to land in a few minutes and I would prefer not to abort it and wait another 5 hours if that's ok. Unfortunately the internal land process makes useful nits like this so much painful ...